components/renku_data_services/data_connectors/migration_utils.py

```diff
@@ -0,0 +1,214 @@
+            target_path=storage.target_path,
+            readonly=storage.readonly,
+        )
+        unsaved_data_connector = models.UnsavedDataConnector(
+            name=storage.name,
+            namespace=project.namespace.slug,
+            slug=data_connector_slug,
+            visibility=project.visibility,
+            created_by="",
+            storage=unsaved_storage,
+        )
+
+        data_connector = await self.data_connector_repo.insert_data_connector(
+            user=user, data_connector=unsaved_data_connector
+        )
+        return data_connector
+
+    async def get_storages_v2(self, requested_by: base_models.APIUser) -> list[storage_models.CloudStorage]:
+        """Get the storages associated with a Renku 2.0 project."""
+        if requested_by.id is None:
+            raise errors.UnauthorizedError(message="You do not have the required permissions for this operation.")
+        if not requested_by.is_admin:
+            raise errors.ForbiddenError(message="Only admins can perform this operation.")
+
+        all_project_ids = await self._get_all_project_ids(requested_by=requested_by)
+        all_project_ids_str = [str(pid) for pid in all_project_ids]
+
+        async with self.session_maker() as session:
+            stmt = select(storage_schemas.CloudStorageORM).where(
+                storage_schemas.CloudStorageORM.project_id.in_(all_project_ids_str)
+            )
+            result = await session.scalars(stmt)
+            storages = result.all()
+            return [storage.dump() for storage in storages]
+
+    async def _delete_storage_v2(self, requested_by: base_models.APIUser, storage_id: ULID) -> ULID | None:
+        """Delete a storage_v2 from the database."""
+        if requested_by.id is None:
+            raise errors.UnauthorizedError(message="You do not have the required permissions for this operation.")
+        if not requested_by.is_admin:
+            raise errors.ForbiddenError(message="Only admins can perform this operation.")
+
+        async with self.session_maker() as session, session.begin():
+            result = await session.scalars(
+                select(storage_schemas.CloudStorageORM).where(storage_schemas.CloudStorageORM.storage_id == storage_id)
+            )
+            storage = result.one_or_none()
+            if storage is None:
+                return None
+            await session.delete(storage)
+            return storage_id
+
+    async def _get_all_project_ids(self, requested_by: base_models.APIUser) -> list[ULID]:
+        """Get all Renku 2.0 projects."""
+        if requested_by.id is None:
+            raise errors.UnauthorizedError(message="You do not have the required permissions for this operation.")
+        if not requested_by.is_admin:
+            raise errors.ForbiddenError(message="Only admins can perform this operation.")
+
+        async with self.session_maker() as session:
+            stmt = select(projects_schemas.ProjectORM)
```

Comment:
nitpick: you don't really need to load the whole objects and dump them, you can just do `select(projects_schemas.ProjectORM).options(load_only(projects_schemas.ProjectORM.id))` to only get a list of IDs and return those.

Reply:
Done in f45f3e4cf8dd80ed3f2cf232a03bdf706dbd6854.